tensorboard/plugins/profile/pod_viewer/pod_viewer_dashboard/pod-viewer-dashboard.ts

```diff
@@ -0,0 +1,232 @@
+      value: () => [],
+      notify: true,
+    },
+    ready_: {
+      type: Boolean,
+      value: false,
+    },
+  },
+  observers: [
+    'dataChanged_(_data, ready_)',
+  ],
+  /**
+   * Updates the UI when new data is loaded.
+   */
+  dataChanged_(newData, ready) {
+    if (!newData || !ready) {
+      return;
+    }
+    this.maxStepId = newData.podStatsSequence.podStatsMap.length - 1;
+    this.curStepId = 0;
+    if (this.maxStepId > 0) {
+      this.podStats = this.populateLowFlopsCompute_(
+        newData.podStatsSequence.podStatsMap['0']);
+    }
+    this.runEnvironment = newData.runEnvironment;
+    this.hloInfoMap = newData.hloInfoMap;
+  },
+  /**
+   * Updates the UI when curStepId changes.
+   */
+  stepChanged_(newStep: number) {
+    if (!this._data) {
+      return;
+    }
+    if (newStep > this.maxStepId) {
+      return;
+    }
+    this.podStats = this.populateLowFlopsCompute_(
+      this._data.podStatsSequence.podStatsMap[newStep.toString()]);
+  },
+  /**
+   * Updates the input of the details card when selected channel changed.
+   */
+  selectedChannelChanged_(newChannel) {
+    if (!newChannel) {
+      return;
+    }
+    this.active = newChannel;
+  },
+  activeBarChartEleChanged_(newEle) {
+    if (!newEle) {
+      return;
+    }
+    this.active = [newEle];
+  },
+  populateLowFlopsCompute_(podStats) {
+    if (!podStats || !this.ready_) return null;
+    let podStatsPerCore = podStats['podStatsPerCore'];
+    for (let i in podStatsPerCore) {
+      let val = podStatsPerCore[i];
+      if (val.hasOwnProperty('lowFlopsComputeUs')) {
+        // already populated.
+        return;
+      }
+      // lowFlopsComputeUs is calculated by deducting all other breakdown from
+      // the total duration.
+      val['lowFlopsComputeUs'] = val.totalDurationUs;
+      for (let j = 0; j < this.stepBreakdownEle.length; j++) {
+        if (j == 1) {
+          continue;
+        }
+        // Skip the lowFlopsComputeUs.
+        val['lowFlopsComputeUs'] -= val[this.stepBreakdownEle[j].key];
+      }
+    }
+   return podStats;
+  },
+  /**
+   * Updates the data sent to stack bar chart when pod stats changed.
+   */
+  podStatsChanged_(newStats) {
+    if (!newStats) {
+      return;
+    }
+    let stepStats = [];
+    for (const i in newStats['podStatsPerCore']) {
+      stepStats.push(newStats['podStatsPerCore'][i]);
+    }
+    stepStats.sort((a, b) => a.chipId - b.chipId);
+    this.stepStats = stepStats;
+    if (newStats['channelDb'].length > 0) {
+      this.channelDb = newStats['channelDb'].sort(
+          (a, b) => b.durationUs - a.durationUs);
+    }
+    if (newStats['allReduceOpDb'].length > 0) {
+      this.allReduceDb = newStats['allReduceOpDb'].sort(
+          (a, b) => b.durationUs - a.durationUs);
+    }
+  },
+  /**
+   * Returns the step number of the current step.
+   */
+  getStepNum(podStats): number {
+    return parseInt(podStats.stepNum, 10);
+  },
+  ready() {
+    this.stepBreakdownEle = [
```

Comment:
Please use appropriate default value in the properties up where the props are defined. Also, I believe doing so will remove the need for property `ready_`

Reply:
Done.